operator/pkg/apis/cassandra/v1beta1/webhook.go

```diff
@@ -55,11 +57,37 @@ func ValidateSingleDatacenter(dc CassandraDatacenter) error {
 			err = attemptedTo("use unsupported Cassandra version '%s'", dc.Spec.ServerVersion)
 		}
 	}
+	if err != nil {
+		return err
+	}
+
+	isDse := dc.Spec.ServerType == "dse"
+	isCassandra3 := dc.Spec.ServerType == "cassandra" && strings.HasPrefix(dc.Spec.ServerVersion, "3.")
+	isCassandra4 := dc.Spec.ServerType == "cassandra" && strings.HasPrefix(dc.Spec.ServerVersion, "4.")
 
-	return err
+	var c map[string]interface{}
+	_ = json.Unmarshal(dc.Spec.Config, &c)
+
+	_, hasJvmOptions := c["jvm-options"]
+	_, hasJvmServerOptions := c["jvm-server-options"]
+	_, hasDseYaml := c["dse-yaml"]
+
+	serverStr := fmt.Sprintf("%s-%s", dc.Spec.ServerType, dc.Spec.ServerVersion)
```

Comment:
Do we want to test any of these new validations in the unit test?

Reply:
All new lines covered!